crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1713,8 +1728,26 @@ unsafe impl<'__w, T: Component> WorldQuery for Mut<'__w, T> {
 }
 
 // SAFETY: access of `Ref<T>` is a subset of `Mut<T>`
-unsafe impl<'__w, T: Component> QueryData for Mut<'__w, T> {
+unsafe impl<'__w, T: Component<Mutability = Mutable>> QueryData for Mut<'__w, T> {
```

Comment:
!!!  Wait, it's currently possible to query for `Mut<T>` on an immutable component?  Good catch!